src/stream_engine.cpp

```diff
@@ -908,9 +910,25 @@ void zmq::stream_engine_t::mechanism_ready ()
             return;
         }
         errno_assert (rc == 0);
-        _session->flush ();
+        flush_session = true;
     }
 
+    if (_options.router_notify & ZMQ_NOTIFY_CONNECT) {
+        msg_t connector;
+        connector.init ();
+        const int rc = _session->push_msg (&connector);
+        if (rc == -1 && errno == EAGAIN) {
+            // See above comment
```

Comment:
Unclear which comment this refers to (in particular when the code is modified in the future)

Reply:
Duplicated the comment for clarification.